packages/compass-aggregations/src/components/aggregation-side-panel/stage-wizard-use-cases/group/group-with-subset.tsx

```diff
@@ -0,0 +1,381 @@
+import {
+  css,
+  Body,
+  Select,
+  Option,
+  spacing,
+  TextInput,
+  ComboboxWithCustomOption,
+} from '@mongodb-js/compass-components';
+import React, { useMemo, useState } from 'react';
+import { connect } from 'react-redux';
+import { ACCUMULATORS as MDB_ACCUMULATORS } from '@mongodb-js/mongodb-constants';
+import type { Document } from 'mongodb';
+import semver from 'semver';
+import {
+  SORT_DIRECTION_OPTIONS,
+  mapFieldsToAccumulatorValue,
+  mapSortDataToStageValue,
+} from '../utils';
+import type { SortDirection } from '../utils';
+import type { RootState } from '../../../../modules';
+
+type AccumulatorName = typeof MDB_ACCUMULATORS[number]['value'];
+type GroupAccumulator = {
+  label: string;
+  value: AccumulatorName;
+  version: string;
+};
+type SubsetAccumulator = GroupAccumulator & {
+  needsSortFields: boolean;
+  nOperator: GroupAccumulator;
+};
+
+const MDB_ACCUMULATORS_VERSION = Object.fromEntries(
+  MDB_ACCUMULATORS.map((x) => [x.value, x.version])
+) as Record<AccumulatorName, string>;
+
+const SUBSET_ACCUMULATORS = [
+  {
+    label: 'First',
+    value: '$first',
+    needsSortFields: false,
+    nOperator: {
+      label: 'FirstN',
+      value: '$firstN',
+    },
+  },
+  {
+    label: 'Last',
+    value: '$last',
+    needsSortFields: false,
+    nOperator: {
+      label: 'LastN',
+      value: '$lastN',
+    },
+  },
+  {
+    label: 'Top',
+    value: '$top',
+    needsSortFields: true,
+    nOperator: {
+      label: 'TopN',
+      value: '$topN',
+    },
+  },
+  {
+    label: 'Bottom',
+    value: '$bottom',
+    needsSortFields: true,
+    nOperator: {
+      label: 'BottomN',
+      value: '$bottomN',
+    },
+  },
+] as const;
+
+const ACCUMULATORS: SubsetAccumulator[] = SUBSET_ACCUMULATORS.map((acc) => {
+  return {
+    ...acc,
+    version: MDB_ACCUMULATORS_VERSION[acc.value],
+    nOperator: {
+      ...acc.nOperator,
+      version: MDB_ACCUMULATORS_VERSION[acc.nOperator.value],
+    },
+  };
+}).filter(Boolean);
```

Comment:
```suggestion
});
```
No harm in keeping it but we probably don't need this filter because the map is always returning an object for every item.

Reply:
Maybe another question - this map generates a list of SUBSET_ACCUMULATORS having a version that is picked from an object. We could do that directly in the array itself, right? something like this?
```
const ACCUMULATORS = [
  {
    label: 'First',
    value: '$first',
    needsSortFields: false,
    version: MDB_ACCUMULATORS_VERSION['$first']
    nOperator: {
      label: 'FirstN',
      value: '$firstN',
      version: MDB_ACCUMULATORS_VERSION['$firstN']
    },
  },
  {
    label: 'Last',
    value: '$last',
    needsSortFields: false,
    version: MDB_ACCUMULATORS_VERSION['$last']
    nOperator: {
      label: 'LastN',
      value: '$lastN',
      version: MDB_ACCUMULATORS_VERSION['$lastN']
    },
  },
// ... other operators
] as const;
```

Reply:
Addressed this in conversation - better typesafety with mapping compared to this